server/controllers/patients/patchPatientData.js

```diff
@@ -0,0 +1,51 @@
+const {
+  patchPatientDataByIdQuery,
+  patientCheckPhone,
+} = require('../../database/queries');
+const { boomify, patientDataValidation } = require('../../utils');
+
+const patchPatientData = async (req, res, next) => {
+  const { firstName, lastName, phone, email, birthday, diseases } = req.body;
+  const { patientId } = req.params;
+
+  try {
+    const {
+      rows: [patient],
+    } = await patientCheckPhone({ phone });
+    if (patient) {
+      return next(boomify(409, 'edit Error', 'phone number is exist'));
+    }
+
+    await patientDataValidation.validate(
+      { firstName, lastName, phone, email, diseases, birthday, patientId },
+      {
+        abortEarly: false,
+      },
+    );
+    const {
+      rows: [data],
+    } = await patchPatientDataByIdQuery(
+      firstName,
+      lastName,
+      phone,
+      email,
+      birthday,
+      diseases,
+      patientId,
+    );
+
+    return res.json({
+      statusCode: 200,
+      message: 'Updated successfully',
+      data,
+    });
+  } catch (error) {
+    return next(
+      error.name === 'ValidationError'
```

Comment:
Because of the parse error , you need to handle the Range Error here.
```js
    if (err.name === 'RangeError') {
      return next(boomify(400, 'RangeError', err.message));
    }
    if (err.name === 'ValidationError') {
      return next(boomify(400, 'Validation Error', err.errors));
    }
    return next(err);
```